testing/tests/DevExpress.ui.widgets.editors/validator.tests.js

```diff
@@ -482,6 +510,67 @@ QUnit.module('Events', {
         assert.strictEqual(params.isValid, true, 'isValid was passed');
         assert.strictEqual(params.brokenRule, null, 'Null should be passed as brokenRule ');
     });
+
+    QUnit.module('Subscription by "on" method', () => {
+        QUnit.test('Validated event should fire', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
+            const handler = sinon.stub();
+
+            const validator = this.fixture.createValidator({
+                name: name,
+                validationRules: [{ type: 'required' }]
+            });
+            validator.on('validated', handler);
+            expectedFailedValidationRule.validator = validator;
+            this.fixture.stubAdapter.getValue.returns(value);
+
+            validator.validate();
+
+            assert.ok(handler.calledOnce, 'Validated handler should be called');
+
+            const params = handler.getCall(0).args[0];
+            assert.ok(handler.calledOn(validator), 'Correct context of action');
+            assert.strictEqual(params.validator, validator, 'Validator reference should be passed');
+            assert.equal(params.value, value, 'Correct value was passed');
+            assert.equal(params.name, name, 'Name of Validator should be passed');
+            assert.strictEqual(params.isValid, false, 'isValid was passed');
+            assert.deepEqual(params.validationRules, [{
+                index: 0,
+                isValid: false,
+                message: 'Login is required',
+                type: 'required',
+                validator: validator,
+                value: value
+            }], 'Correct rules were passed');
+            assert.deepEqual(params.brokenRule, expectedFailedValidationRule, 'Failed rules were passed');
+        });
+
+        QUnit.test('Validated event should fire correctly after option runtime change', function(assert) {
+            const value = '';
+            const name = 'Login';
+            const expectedFailedValidationRule = { index: 0, type: 'required', isValid: false, message: 'Login is required', validator: {}, value: value };
+            const handler = sinon.stub();
+            const handlerAfterChange = sinon.stub();
+
+            const validator = this.fixture.createValidator({
+                name: name,
+                validationRules: [{ type: 'required' }]
+            });
+
+            expectedFailedValidationRule.validator = validator;
+            this.fixture.stubAdapter.getValue.returns(value);
+
+            validator.on('validated', handler);
+            validator.validate();
+            assert.ok(handler.calledOnce, 'Validated handler should be called');
```

Comment:
We already have this assert here:
https://github.com/DevExpress/DevExtreme/pull/11369/files#diff-dc05fa7fdaba59cd05c488434aca063dR531